### DISCUSSION TITLE
Lifetime of silero-VAD v4.0

### DISCUSSION THREAD

**Discussion by ArianAzg** [snakers4/silero-vad#512] [Q&A] [answered]
Hello Silero Team,

Thank you all for your wonderful work on this voice activity detection. I am wondering if the version 4 of your VAD (with dynamic input window size) remains active or its lifetime will reach to an end in the future? 
I am using it as part of  one my projects and it performs well on noisy data. Obviously, I will acknowledge all your invaluable works in my future publications. 

Thank you very much!
Sincerely,
Arian

**Comment by snakers4** [comment, marked as answer]
Hi,

Older versions can be obtained via repository tags - please see https://github.com/snakers4/silero-vad/issues/474

>  remains active or its lifetime will reach to an end in the future?

We are not active maintaining the legacy models, but they will keep working with respective versions of PyTorch and ONNX-runtime, and most likely with new versions too.

**Comment by x86Gr** [comment]
Since version 3 4 and 5 have different strengths with different domains, will it be possible to switch between the models, or should just keep three separated python VENVs?

**Reply by snakers4** [reply]
You can try this  https://github.com/snakers4/silero-vad/issues/474
